lib/launchers/openstack10.rb

```diff
@@ -687,6 +687,29 @@ def assign_ip(instance)
       end
     end
 
+    def allocate_floating_ip(network_name)
+      request_url = self.os_network_url + "/v2.0/floatingips"
+      method = "POST"
+      network = get_networks.find {|n| n["name"] == network_name}
```

Comment:
If you only allocate floating IPs, then use method `#floating_ip_networks` instead of `#get_networks`. If you can allocate IPs from networks that are NOT `router:external`, then you can rename the method to `allocate_ip` instead of `allocate_floating_ip`.

Reply:
ok, got it, will update later